Sources/Reporters/DirectoryTree/DirectoryTreeReporter.swift

```diff
@@ -7,54 +7,57 @@
 
 import Foundation
 
-/// Reports a directory tree structure for the given directory urls.
-public struct DirectoryTreesReporter: DiagnosticsReporting {
-    let title: String = "Directory Storage Trees"
-    let directories: [URL]
+public struct Directory {
+    let url: URL
+    let customisedName: String?
     let maxDepth: Int
     let maxLength: Int
     let includeHiddenFiles: Bool
     let includeSymbolicLinks: Bool
     let printFullPath: Bool
-
-    /// Creates a new reporter for directory trees.
-    /// - Parameters:
-    ///   - directories: The directories to create trees for.
-    ///   - maxDepth: The max depth of directories to visit. Defaults to `.max`.
-    ///   - maxLength: The max length of nodes in a directory to handle. Defaults to `10`.
-    ///   - includeHiddenFiles: Whether hidden files should be captured. Defaults to `false`.
-    ///   - includeSymbolicLinks: Whether symbolic links should be captured. Defaults to `false`.
-    ///   - printFullPath: Whether the full path of the node should be printed or just the name. Defaults to `false`.
-    public init(
-        directories: [URL],
-        maxDepth: Int = .max,
-        maxLength: Int = 10,
-        includeHiddenFiles: Bool = false,
-        includeSymbolicLinks: Bool = false,
-        printFullPath: Bool = false
-    ) {
-        self.directories = directories
+    init(url: URL,
+         customisedName: String? = nil,
+         maxDepth: Int = .max,
+         maxLength: Int = 10,
+         includeHiddenFiles: Bool = false,
+         includeSymbolicLinks: Bool = false,
+         printFullPath: Bool = false) {
+        self.url = url
+        self.customisedName = customisedName
         self.maxDepth = maxDepth
         self.maxLength = maxLength
         self.includeHiddenFiles = includeHiddenFiles
         self.includeSymbolicLinks = includeSymbolicLinks
         self.printFullPath = printFullPath
     }
+}
+
+/// Reports a directory tree structure for the given directory urls.
+public struct DirectoryTreesReporter: DiagnosticsReporting {
+    let title: String = "Directory Storage Trees"
+    let directories: [Directory]
+
+    /// Creates a new reporter for directory trees.
+    /// - Parameters:
+    ///   - directories: The directories to create trees for.
+    public init(directories: [Directory]) {
```

Comment:
Can we add a deprecated init using the previous initializer that will reuse this initializer with a single `Directory`? That way, this change won't be breaking, and we allow users to slowly adopt.

Reply:
I like this!

Reply:
Thanks :)  @BasThomas  & @AvdLee  for your inputs here. I'll complete my PR with your suggested changes, while also keeping in mind the customised Directory name feature instead of displaying the UUID and also few more potential worthy changes.